tls/s2n_change_cipher_spec.c

```diff
@@ -17,25 +17,53 @@
 
 #include "error/s2n_errno.h"
 
+#include "tls/s2n_cipher_suites.h"
 #include "tls/s2n_connection.h"
 #include "tls/s2n_tls.h"
 
 #include "stuffer/s2n_stuffer.h"
 
 #include "utils/s2n_safety.h"
 
-/* From RFC5246 7.1. */
+/* From RFC5246 7.1: https://tools.ietf.org/html/rfc5246#section-7.1 */
 #define CHANGE_CIPHER_SPEC_TYPE  1
 
-int s2n_server_ccs_recv(struct s2n_connection *conn)
+int s2n_change_cipher_spec_recv(struct s2n_connection *conn)
 {
     uint8_t type;
 
     GUARD(s2n_stuffer_read_uint8(&conn->handshake.io, &type));
     S2N_ERROR_IF(type != CHANGE_CIPHER_SPEC_TYPE, S2N_ERR_BAD_MESSAGE);
 
+    return 0;
+}
+
+int s2n_client_ccs_recv(struct s2n_connection *conn)
+{
+    GUARD(s2n_change_cipher_spec_recv(conn));
+
+    /* Zero the sequence number */
+    struct s2n_blob seq = {.data = conn->secure.client_sequence_number,.size = sizeof(conn->secure.client_sequence_number) };
+    GUARD(s2n_blob_zero(&seq));
+
+    /* Compute the finished message */
+    GUARD(s2n_prf_client_finished(conn));
+
+    /* Update the client to use the cipher-suite */
+    conn->client = &conn->secure;
+
+    /* Flush any partial alert messages that were pending */
+    GUARD(s2n_stuffer_wipe(&conn->alert_in));
```

Comment:
I know you were just moving this function, but I'm quite confused why this line was ever added.

I don't think we want to swallow an alert here. Maybe I'm missing something obvious.

Reply:
Looks like it's been that way since the beginning, part of Colm's [first import](https://github.com/awslabs/s2n/commit/719eccd786625945bc42f732903e2a3e255c6bad). My guess at 'why' would be that if we receive the ccs message, it means that the handshake was successful, so any alerts we were sent didn't affect completing the handshake so can be ignored? But Colm may know more.

Reply:
From Colm: https://www.mitls.org/pages/attacks/Alert I'll add a comment to make this clearer in the future.